tests/circuit_breaking_test.py

```diff
@@ -0,0 +1,214 @@
+                                    "matchRules": [
+                                        {
+                                            "fullPathMatch": "/grpc.testing.TestService/EmptyCall"
+                                        }
+                                    ],
+                                    "service": self.td.alternative_backend_service.url,
+                                },
+                            ],
+                        }
+                    ],
+                }
+            )
+
+        with self.subTest("03_create_target_proxy"):
+            self.td.create_target_proxy()
+
+        with self.subTest("04_create_forwarding_rule"):
+            self.td.create_forwarding_rule(self.server_xds_port)
+
+        default_test_servers: List[_XdsTestServer]
+        alternate_test_servers: List[_XdsTestServer]
+        with self.subTest("05_start_test_servers"):
+            default_test_servers = self.startTestServers()
+            alternate_test_servers = self.startTestServers(
+                server_runner=self.alternate_server_runner
+            )
+
+        with self.subTest("06_add_server_backends_to_backend_services"):
+            self.setupServerBackends()
+            # Add backend to alternative backend service
+            (
+                neg_name_alt,
+                neg_zones_alt,
+            ) = self.alternate_k8s_namespace.parse_service_neg_status(
+                self.alternate_server_runner.service_name, self.server_port
+            )
+            self.td.alternative_backend_service_add_neg_backends(
+                neg_name_alt, neg_zones_alt
+            )
+
+        with self.subTest("07_patch_backends_with_circuit_breakers"):
+            self.td.backend_service_patch_backends(
+                circuit_breaker={"maxRequests": _INITIAL_UNARY_MAX_REQUESTS}
+            )
+            self.td.alternative_backend_service_patch_backends(
+                circuit_breaker={"maxRequests": _INITIAL_EMPTY_MAX_REQUESTS}
+            )
+
+        test_client: _XdsTestClient
+        with self.subTest("08_start_test_client"):
+            test_client = self.startTestClient(
+                default_test_servers[0], rpc="UnaryCall,EmptyCall", qps=_QPS
+            )
+
+        with self.subTest("09_test_client_xds_config_exists"):
+            self.assertXdsConfigExists(test_client)
+
+        with self.subTest("10_test_server_received_rpcs_from_test_client"):
+            self.assertRpcsEventuallyGoToGivenServers(
+                test_client, default_test_servers + alternate_test_servers
+            )
+
+        with self.subTest("11_configure_client_with_keep_open"):
+            test_client.update_config.configure(
+                rpc_types=["UNARY_CALL", "EMPTY_CALL"],
+                metadata={
+                    ("UNARY_CALL", "rpc-behavior", "keep-open"),
+                    ("EMPTY_CALL", "rpc-behavior", "keep-open"),
+                },
```

Comment:
Added constants recently to distinguish from the other "syntax" we use: `EmptyCall`, `UnaryCall`.

Note - we'll need to do `from framework.rpc import grpc_testing`  at the top and use constants in the following subtests.

```suggestion
                rpc_types=grpc_testing.RPC_TYPES_BOTH_CALLS,
                metadata={
                    (
                        grpc_testing.RPC_TYPE_UNARY_CALL,
                        "rpc-behavior",
                        "keep-open",
                    ),
                    (
                        grpc_testing.RPC_TYPE_EMPTY_CALL,
                        "rpc-behavior",
                        "keep-open",
                    ),
                },
```